### ISSUE TITLE
build.cmd fails native build on a virtual machine

### ISSUE THREAD

**Issue by davmason** [dotnet/coreclr#14605]
I do my development in a VM hosted through hyper-v. On this VM, WMI does not return a value for NumberOfEnabledCore. #14509 introduced a dependency on it for the native build.

```
D:\coreclr>wmic cpu get NumberOfEnabledCore /value


NumberOfEnabledCore=




D:\coreclr>
```

This causes the recent build.cmd changes to fail with a bunch of these errors:
```
C:\Program Files (x86)\Microsoft Visual Studio\Preview\Enterprise\Common7\IDE\VC\VCTargets\Microsoft.CppCommon.targets(
" is an invalid value for the "ProcessorNumber" parameter of the "CL" task. The "ProcessorNumb
er" parameter is of type "System.Int32". [D:\coreclr\bin\obj\Windows_NT.x64.Debug\src\coreclr\hosts\coreconsole\CoreCon
sole.vcxproj]
```

It seems like NumberOfEnabledCore isn't guaranteed to have a value, so an extra check is necessary with a default value for when it doesn't exist.

FYI - I worked around it for now by reverting the change, and am not blocked - but I imagine I can't be the only person building on a VM.

/CC @4creators

**Comment by 4creators** [comment]
It is fixed in #14578 since CL parallelism control is moved to CMake script. Actually I have verified that NumberOfEnabledCore value never returns any value on VM guest so this is really a problem - it is undoceumented in MSDN/docs as well.

**Comment by danmosemsft** [comment]
@davmason can you close this then?

**Comment by 4creators** [comment]
It should be resolved now as #14515 has been merged.
